### PR TITLE
Test and CI updates

### PR DESCRIPTION
**Pull request by kanderso-nrel**
## Description

This PR:
- Drops python 3.6 and adds python 3.9, 3.10 to CI tests
- Some tests currently use deprecated pvlib functions.  This PR adds equivalent tests using non-deprecated pvlib functions.  Each set of tests is marked to only run if the appropriate pvlib version is installed.  The older tests can be deleted when we eventually drop support for `pvlib<0.9.0` down the line. 
- Resolves a few pandas deprecation warnings in the test functions.  Most of the pandas warnings are not easily fixed and are not addressed here.

Because this PR doesn't add any new tests using *multiple* Arrays, it only partially addresses #116.

## Checklist

- ~[ ] Closes #xxx~
- ~[ ] Added new API functions to `docs/api.rst`~
- ~[ ] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings~
- [x] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- ~[ ] Non-API functions clearly documented with docstrings or comments as necessary~
- ~[ ] Added tests to cover all new or modified code~
- [x] Pull request is nearly complete and ready for detailed review

### CODE REVIEW

**Review by cwhanse — APPROVED** [review]

**Comment by cwhanse** [comment]
OK to merge when the whatsnew conflict is resolved.

**Comment by cwhanse** [comment]
Build fails after merging #127. But it appears the failures will be fixed with this PR.

**Comment by kanderso-nrel** [comment]
> Build fails after merging https://github.com/pvlib/pvanalytics/pull/127. But it appears the failures will be fixed with this PR.

Tests here are green, so I expect so.

**Pull request merged**
